src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,96 @@
+package core.basesyntax.service;
+
+import static core.basesyntax.db.Storage.people;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "user123";
+    private static final String INVALID_LOGIN = "mkwvs";
+    private static final String VALID_PASSWORD = "password123";
+    private static final String INVALID_PASSWORD = "nvdf";
+    private static final int VALID_AGE = 18;
+    private static final int INVALID_AGE = 17;
+    private static final int MIN_AGE = 18;
+    private static final String NULL_LOGIN = null;
+    private static final String NULL_PASSWORD = null;
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+    private User validUser;
+
+    @BeforeEach
+    void setUp() {
+        people.clear();
+        validUser = new User();
+        validUser.setLogin(VALID_LOGIN);
+        validUser.setPassword(VALID_PASSWORD);
+        validUser.setAge(VALID_AGE);
+    }
+
+    @Test
+    void registerNullUserThrowRegistrationException_notOk() {
+        validUser = null;
+        RegistrationException exception = assertThrows(RegistrationException.class,
+                () -> registrationService.register(validUser));
+        assertEquals("User can't be null", exception.getMessage());
+    }
+
+    @Test
+    public void registerValidUser_Ok() {
+        User registeredUser = registrationService.register(validUser);
+        assertEquals(validUser, registeredUser);
+    }
+
+    @Test
+    void registerAlreadyAddedUserThrowRegistrationException_notOk() {
+        User registeredUser = registrationService.register(validUser);
```

Comment:
check common mistakes, we shouldn't use `register()` for adding the first user to the storage, because this method is under a test, and we can not guarantee that it is working properly, thus it can give us a false-positive result. Instead we should use `storage` or `storageDao`